Lib/test/test_ast.py

```diff
@@ -607,6 +608,75 @@ def test_issue18374_binop_col_offset(self):
         self.assertEqual(grandchild_binop.end_col_offset, 3)
         self.assertEqual(grandchild_binop.end_lineno, 1)
 
+class ASTCompareTest(unittest.TestCase):
+    def test_normal_compare(self):
+        self.assertEqual(ast.parse("x = 10"), ast.parse("x = 10"))
+        self.assertNotEqual(ast.parse("x = 10"), ast.parse(""))
+        self.assertNotEqual(ast.parse("x = 10"), ast.parse("x"))
+        self.assertNotEqual(ast.parse("x = 10;y = 20"), ast.parse("class C:pass"))
+
+    def test_literals_compare(self):
+        self.assertEqual(ast.Num(), ast.Num())
+        self.assertEqual(ast.Num(-20), ast.Num(-20))
+        self.assertEqual(ast.Num(10), ast.Num(10))
+        self.assertEqual(ast.Num(2048), ast.Num(2048))
+        self.assertEqual(ast.Str(), ast.Str())
+        self.assertEqual(ast.Str("ABCD"), ast.Str("ABCD"))
+        self.assertEqual(ast.Str("中文字"), ast.Str("中文字"))
+
+        self.assertNotEqual(ast.Num(10), ast.Num(20))
```

Comment:
can you add a testcase for `ast.Num(10)` and `ast.Num(10.0)` -- I suspect the current implementation will return `True` but perhaps it should not

The same for `ast.Num(1)` and `ast.NamedConstant(True)` is probably also a good test

Reply:
@asottile Current implementation returns False, so that's all good

Reply:
awesome, just wanted to make sure that's captured in the tests

Reply:
@asottile wait hold on, I added it in the tests and it passed locally when I tried it in my re-compiled 3.9 version, but on the CI it does not. I guess it will need some fixing after all

Reply:
This issue has been resolved.